pkg/pushsync/pushsync.go

```diff
@@ -361,33 +336,25 @@ func (ps *PushSync) pushToClosest(ctx context.Context, ch swarm.Chunk, origin bo
 		}
 	}
 
-	retry()
-
-	// nextPeer attempts to lookup the next peer to push the chunk to, if there are overdrafted peers the boolean would signal a re-attempt
-	nextPeer := func() (peer swarm.Address, err error) {
-
-		fullSkipList := append(skip.ChunkPeers(ch.Address()), ps.skipList.ChunkPeers(ch.Address())...)
-
-		peer, err = ps.topologyDriver.ClosestPeer(ch.Address(), includeSelf, topology.Filter{Reachable: true}, fullSkipList...)
-		if err != nil {
-			if errors.Is(err, topology.ErrWantSelf) {
-				if !ps.warmedUp() {
-					return swarm.ZeroAddress, ErrWarmup
-				}
+	nextPeer := func() (swarm.Address, error) {
 
-				if !ps.radiusChecker.IsWithinStorageRadius(ch.Address()) {
-					return swarm.ZeroAddress, ErrOutOfDepthStoring
-				}
-
-				ps.pushToNeighbourhood(ctx, fullSkipList, ch, origin, originAddr)
-				return swarm.ZeroAddress, err
-			}
+		// we are a full node and in the neighborhood of the chunk
+		if ps.fullNode && ps.radiusChecker.IsWithinStorageRadius(ch.Address()) {
+			return swarm.ZeroAddress, topology.ErrWantSelf
+		}
 
-			return swarm.ZeroAddress, fmt.Errorf("closest peer: %w", err)
+		// find closer peer
+		peer, err := ps.topologyDriver.ClosestPeer(ch.Address(), ps.fullNode, topology.Filter{Reachable: true}, ps.skipList.ChunkPeers(ch.Address())...)
+		// only return ErrWantSelf if we are in the neighborhood of the chunk
+		if errors.Is(err, topology.ErrWantSelf) {
+			return swarm.ZeroAddress, topology.ErrNotFound
 		}
-		return peer, nil
+
+		return peer, err
 	}
 
+	retry()
+
 	for sentErrorsLeft > 0 {
```

Comment:
does `sentErrorsLeft` needs to be increased to accommodate errors for all forwarding request in case of multiplexing to entire neighborhood?